### PR TITLE
Rename the role - replacing '-' with '_'

### PR DESCRIPTION
**Pull request by irosenzw**
Rename the role along with its RPM

Signed-off-by: Ido Rosenzwig <irosenzw@redhat.com>

### CODE REVIEW

**Comment by tiraboschi** [comment]
We also have to:
- [ ] add role_name in meta/main.yml
- [ ] add obsoletes in the spec file
- [ ] a couple of lines like https://github.com/oVirt/ovirt-ansible-hosted-engine-setup/blob/master/build.sh#L47 in build.sh to create a symlink at build time for backward compatibility with ovirt.hosted-engine-setup -> ovirt.hosted_engine_setup
- [ ] fix all the example playbooks using the new role name
- [ ] fix README.md using the new role name

File: README.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ All the playbooks can be found inside the `examples/` folder.
     ovirt_engine_setup_offline: true
     ovirt_engine_setup_admin_password: "{{ he_admin_password }}"
   roles:
-    - role: oVirt.engine-setup
+    - role: ovirt.engine_setup
```
Comment (tiraboschi) [inline README.md]: I think we are still no ready for this, please keep the old name for now

File: examples/hosted_engine_deploy_localhost.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@
     ovirt_engine_setup_offline: true
     ovirt_engine_setup_admin_password: "{{ he_admin_password }}"
   roles:
-    - role: oVirt.engine-setup
+    - role: ovirt.engine_setup
```
Comment (tiraboschi) [inline examples/hosted_engine_deploy_localhost.yml]: Same

File: examples/hosted_engine_deploy_remotehost.yml [hunk 3]
```diff
@@ -38,7 +38,7 @@
     ovirt_engine_setup_offline: true
     ovirt_engine_setup_admin_password: "{{ he_admin_password }}"
   roles:
-    - role: oVirt.engine-setup
+    - role: ovirt.engine_setup
```
Comment (tiraboschi) [inline examples/hosted_engine_deploy_remotehost.yml]: Same

File: build.sh [hunk 4]
```diff
@@ -29,9 +30,9 @@ dist() {
    -e "s|@RPM_RELEASE@|$RPM_RELEASE|g" \
    -e "s|@PACKAGE_NAME@|$PACKAGE_NAME|g" \
    -e "s|@PACKAGE_VERSION@|$PACKAGE_VERSION|g" \
-   < ovirt-ansible-hosted-engine-setup.spec.in > ovirt-ansible-hosted-engine-setup.spec
+   < ovirt_ansible_hosted_engine_setup.spec.in > ovirt_ansible_hosted_engine_setup.spec
```
Comment (tiraboschi) [inline build.sh]: Still not 100% sure about the rpm name change. Everything else is already ovirt-

**Comment by irosenzw** [comment]
@tiraboschi please review

**Pull request merged**